From 84458195be8ac541f50635996b7c4f6931d5fe2d Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 2 Mar 2022 18:20:29 +0100
Subject: [PATCH] ARM: dts: stm32: Add OPTEE interrupts to stm32mp131

Describes optee IRQ chip that is used to provide some interrupts
generated by OP-TEE. Adds a phandle to the optee for its interrupts
consumers.

Change-Id: I68bdad2a068e8afce8e9a456179f532079d5cadd
Signed-off-by: Pascal Paillet <p.paillet@st.com>
---
 arch/arm/boot/dts/st/stm32mp131.dtsi | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/st/stm32mp131.dtsi b/arch/arm/boot/dts/st/stm32mp131.dtsi
index 97f589c15696..f1fa3e3710eb 100644
--- a/arch/arm/boot/dts/st/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp131.dtsi
@@ -40,11 +40,13 @@ arm_wdt: watchdog {
 	};
 
 	firmware {
-		optee {
+		optee: optee {
 			method = "smc";
 			compatible = "linaro,optee-tz";
 			interrupt-parent = <&intc>;
 			interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
 		};
 
 		scmi: scmi {
-- 
2.39.5

